### PR TITLE
Update to Mirage 4.5.0

### PR DESCRIPTION
**Pull request by palainp**
Dear devs,
This PR wants to update to recent mirage API. In doing so it appears that `Memory.quick_stat()` reports wrong values (~~I haven't investigated why yet, and maybe it was present before, but now it goes lower than the limits in `memory_pressure.ml`~~ EDIT: There was a bad computation in `ocaml-solo5` again):
```
Memory.quick_stat() INF [memory_pressure] Writing meminfo: free 11MiB / 27MiB (39.99 %)
Memory.stat() INF [memory_pressure] Writing meminfo: free 19MiB / 27MiB (70.31 %)
```
~~With wrong values, the firewall starts to drop packets (`Memory_critical` being returned by `Memory_pressure.status()`). So I've replaced `Memory.quick_stat()` with `Memory.stat()`, but this is at the cost of going through the whole heap and `Memory_pressure.status` calls it for each packets. But I'd rather the firewall be slower than unusable :)~~

~~Going further will need to dive more into `ocaml-solo5/nolibc/malloc.i` as the bug seems to come from there but the code is really hard :'(~~
Best,

### CODE REVIEW

**Comment by palainp** [comment]
With https://github.com/mirage/ocaml-solo5/pull/135 using `quick_stat()` is ok now so I removed the last modifications to have it back :)

**Comment by palainp** [comment]
Will be included in #178 .

File: config.ml [hunk 1]
```diff
@@ -1,37 +1,32 @@
+(* mirage >= 4.5.0 *)
```
Comment (hannesm) [inline config.ml:1]: & < 5.0 (from what I'd expect) ;)

File: config.ml [hunk 2]
```diff
@@ -1,37 +1,32 @@
...
       package "mirage-net-xen";
       package "ipaddr" ~min:"5.2.0";
       package "mirage-qubes" ~min:"0.9.1";
-      package ~min:"3.0.1" "mirage-nat";
+      package "mirage-nat" ~min:"3.0.1";
       package "mirage-logs";
       package "mirage-xen" ~min:"8.0.0";
-      package ~min:"6.4.0" "dns-client";
+      package "dns-client" ~min:"6.4.0";
       package "pf-qubes";
```
Comment (hannesm) [inline config.ml:30]: are these argument reorderings needed?

File: memory_pressure.ml [hunk 3]
```diff
@@ -26,6 +25,7 @@ let meminfo stats =
                   Cached: 0 kB\n\
                   SwapTotal: 0 kB\n\
                   SwapFree: 0 kB\n" (mem_total / 1024) (mem_free / 1024)
+*)
```
Comment (hannesm) [inline memory_pressure.ml:28]: let's remove unused code -- there's git for the archeologists

**Comment by palainp** [comment]
Sorry I closed while you are reviewing it, I'll add your changes in #178.

**Comment by hannesm** [comment]
@palainp thanks. no worries. :)

File: config.ml (hunk 2 above)
Comment (palainp) [reply config.ml:30]: I don't think it needed, this was just cosmetic to keep packages aligned.

File: config.ml (hunk 1 above)
Comment (palainp) [reply config.ml:1]: yes!

File: memory_pressure.ml (hunk 3 above)
Comment (palainp) [reply memory_pressure.ml:28]: that's right :)

**Pull request closed without merging**